rpadmin/admin.go

```diff
@@ -536,7 +537,18 @@ func (a *AdminAPI) sendAll(rootCtx context.Context, method, path string, body, i
 			// Only one request should be successful, but for
 			// paranoia, we guard keeping the first successful
 			// response.
-			once.Do(func() { resURL, res = myURL, myRes })
+			mutex.Lock()
+			defer mutex.Unlock()
+
+			if resURL != "" {
```

Comment:
Kinda seems like moving `maybeUnmarshalRespInto` into this closure would make everything _much_ easier to think about. (That and an [errgroup](https://pkg.go.dev/golang.org/x/sync/errgroup) would be much better suited than a multierrgroup but I'll pick one hill to die on here...)

```go
var succeeded atomic.Bool

// ...

defer myRes.Body.Close()

if !succeeded.CompareAndSwap(false, true) {
   // Someone else beat us.  
  return nil
}

// Unmarshal the successful call.
return maybeUnmarshalRespInto(method, resURL, res, into)

// ... Out of closure

if succeeded.Load() {
  // Success or error un unmarshal
}
return grp.Errs()
```